.github/actions/deploy-dataflow-pipeline/action.yml

```diff
@@ -0,0 +1,123 @@
+    required: true
+  template_metadata_file:
+    description: 'The local file containing the template metadata defining the pipeline.'
+    required: true
+  template_file_name:
+    description: 'The name of the compiled template file that will be stored in GCS.'
+    required: true
+  project:
+    description: 'The GCP Project to use.'
+    required: true
+  region:
+    description: 'The GCP region to use.'
+    required: true
+  bucket_name:
+    description: 'The gsc bucket to store the template file.'
+    required: true
+  network_name:
+    description: 'The name of GCP network to use.'
+    required: true
+  subnet_name:
+    description: 'Then name of the subnet to use.'
+    required: false
+  pipeline_name:
+    description: 'The name of pipeline.'
+    required: true
+  pipline_sdk:
+    description: 'SDK language of the pipeline. Must be one of: JAVA, PYTHON, GO.'
+    required: true
+  pipeline_type:
+    description: 'The type of pipeline. Must be one of: BATCH, STREAMING.'
+    required: true
+  pipeline_schedule:
+    description: 'The cron formatted schedule for the pipeline.'
+    required: true
+  pipeline_parameters:
+    description: 'The command line parameters for the pipeline.'
+    required: false
+
+
+runs:
+  using: 'composite'
+  steps:
+    - name: 'Checkout'
+      uses: 'actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11' # ratchet:actions/checkout@v4
+
+    - id: 'auth'
+      name: 'Authenticate to Google Cloud'
+      uses: 'google-github-actions/auth@35b0e87d162680511bf346c299f71c9c5c379033' # ratchet:google-github-actions/auth@v1
+      with:
+        workload_identity_provider: '${{ inputs.wif_provider }}'
+        service_account: '${{ inputs.wif_service_account }}'
+
+    - name: 'Setup gcloud'
+      uses: 'google-github-actions/setup-gcloud@e30db14379863a8c79331b04a9969f4c1e225e0b' # ratchet:google-github-actions/setup-gcloud@v1
+      with:
+        install_components: 'beta'
+
+    - name: 'Deploy Dataflow Pipeline'
+      shell: 'bash'
+      env:
+        IMAGE: '{{ inputs.image }}'
+        TEMPLATE_METADATA_FILE: '{{ inputs.template_metadata_file }}'
+        TEMPLATE_FILE_NAME: '{{ inputs.template_file_name }}'
+        PROJECT: '{{ inputs.project }}'
+        REGION: '{{ inputs.region }}'
+        NETWORK_NAME: '${{ inputs.network_name }}'
+        SUBNET_NAME: '${{ inputs.subnet_name }}'
+        BUCKET_NAME: '${{ inputs.bucket_name }}'
+        PIPELINE_NAME: '${{ inputs.pipeline_name }}'
+        PIPELINE_SDK: '${{ inputs.pipeline_sdk }}'
+        PIPELINE_TYPE: '${{ inputs.pipeline_type }}'
+        PIPELINE_SCHEDULE: '${{ inputs.pipeline_schedule }}'
+        PIPELINE_PARAMETERS: '${{ inputs.pipeline_parameters }}'
+      run: |-
+        GCS_TEMPLATE_FILE="gs://${BUCKET_NAME}/templates/${TEMPLATE_FILE_NAME}"
+
+        gcloud dataflow flex-template build "${GCS_TEMPLATE_FILE}" \
+          --image "${IMAGE}" \
```

Comment:
This should accept a `--project` flag, we should always try to use it, otherwise, things default to the service account project (which is bad). Can we add it just to be certain?

Reply:
done. I didn't realize `--project` was a global flag for any `gcloud` command. Definitely makes sense to add it in that context.